modules/ballerina-builtin/src/main/java/org/ballerinalang/nativeimpl/task/ScheduleAppointment.java

```diff
@@ -0,0 +1,97 @@
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.natives.AbstractNativeFunction;
+import org.ballerinalang.natives.annotations.Argument;
+import org.ballerinalang.natives.annotations.Attribute;
+import org.ballerinalang.natives.annotations.BallerinaAnnotation;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.ReturnType;
+import org.ballerinalang.util.codegen.cpentries.FunctionRefCPEntry;
+
+/**
+ * Native function ballerina.model.task:scheduleAppointment.
+ */
+@BallerinaFunction(
+    packageName = "ballerina.task",
+    functionName = "scheduleAppointment",
+    args = {@Argument(name = "onTrigger", type = TypeKind.ANY),
+            @Argument(name = "onError", type = TypeKind.ANY),
+            @Argument(name = "appointmentScheduler", type = TypeKind.STRUCT, structType = "AppointmentScheduler")},
+    returnType = {@ReturnType(type = TypeKind.INT), @ReturnType(type = TypeKind.ANY)},
+    isPublic = true
+)
+@BallerinaAnnotation(annotationName = "Description", attributes = {@Attribute(name = "value",
+        value = "Schedules the task service with cron expression") })
+@BallerinaAnnotation(annotationName = "Param", attributes = {@Attribute(name = "onTrigger",
+        value = "The schedule function as any type") })
+@BallerinaAnnotation(annotationName = "Param", attributes = {@Attribute(name = "onError",
+        value = "The error function as any type") })
+@BallerinaAnnotation(annotationName = "Param", attributes = {@Attribute(name = "appointmentScheduler",
+        value = "The struct with required attributes") })
+@BallerinaAnnotation(annotationName = "Return", attributes = {@Attribute(name = "int)",
+        value = "The identifier of the task") })
+@BallerinaAnnotation(annotationName = "Return", attributes = {@Attribute(name = "any)",
+        value = "The error which is occurred while scheduling the task") })
+public class ScheduleAppointment extends AbstractNativeFunction {
+    private static final Log log = LogFactory.getLog(ScheduleAppointment.class.getName());
+
+    public BValue[] execute(Context ctx) {
+        FunctionRefCPEntry onTriggerFunctionRefCPEntry = null;
+        FunctionRefCPEntry onErrorFunctionRefCPEntry = null;
+        if (ctx.getControlStackNew().getCurrentFrame().getRefLocalVars()[0] != null && ctx.getControlStackNew()
+                .getCurrentFrame().getRefLocalVars()[0] instanceof BFunctionPointer) {
+            onTriggerFunctionRefCPEntry = ((BFunctionPointer) getRefArgument(ctx, 0)).value();
+        }
+        if (ctx.getControlStackNew().getCurrentFrame().getRefLocalVars()[1] != null && ctx.getControlStackNew()
+                .getCurrentFrame().getRefLocalVars()[1] instanceof BFunctionPointer) {
+            onErrorFunctionRefCPEntry = ((BFunctionPointer) getRefArgument(ctx, 1)).value();
+        }
+        BValue scheduler = getRefArgument(ctx, 2);
+        long minute = ((BStruct) scheduler).getIntField(0);
```

Comment:
It would be better to cast to BStruct only once, and use that.
eg:  
``` java
BStruct scheduler = (BStruct) getRefArgument(ctx, 2);
long minute = scheduler.getIntField(0);
...
```